inc/telemetry/namespace.php

```diff
@@ -0,0 +1,445 @@
+		'created' => $current_user->user_registered,
+		'avatar' => get_avatar_url( $current_user->ID ),
+	];
+
+	/**
+	 * Filter user traits delivered to segment.io.
+	 *
+	 * @param array $traits User traits to send to Segment.io.
+	 */
+	$traits = (array) apply_filters( 'altis.telemetry.user_traits', $traits );
+
+	return [
+		'id' => $id,
+		'opt_in' => true,
+		'traits' => $traits,
+	];
+}
+
+/**
+ * Get a unique ID for the current (local) environment.
+ *
+ * Generates a persistent, unique ID for the current local environment.
+ *
+ * @return string|null Null for non-local environments, unique ID otherwise.
+ */
+function get_local_install_id() : ?string {
+	if ( Altis\get_environment_type() !== 'local' ) {
+		return null;
+	}
+
+	$id = get_network_option( null, 'altis_tracking_id', null );
+	if ( ! empty( $id ) ) {
+		return $id;
+	}
+
+	// Hash the registration date of the first site to get a unique ID.
+	$site = get_blog_details( get_main_site_id(), false );
+	$generated = 'local-' . substr( hash( 'sha1', $site->registered ), 0, 6 );
+	update_network_option( null, 'altis_tracking_id', $generated );
+
+	return $generated;
+}
+
+/**
+ * Get current Altis environment details.
+ *
+ * @return array
+ */
+function get_environment_details() : array {
+	$id = Altis\get_environment_name();
+	$type = Altis\get_environment_type();
+	if ( $type === 'local' ) {
+		// Create distinct_id (important for Mixpanel).
+		$id = get_local_install_id();
+	}
+
+	return [
+		'id' => $id,
+		'traits' => [
+			'environment' => $type,
+			'domain' => get_site_url(),
+			'feature_tier' => Altis\get_feature_tier(),
+			'environment_tier' => Altis\get_environment_tier(),
+			'support_tier' => Altis\get_support_tier(),
+		],
+	];
+}
+
+/**
+ * Load segment.io JS.
+ *
+ * @return void
+ */
+function load_segment_js() {
+	?>
+	<script>
+		// Segment - Load
+		!function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(e){return function(){var t=Array.prototype.slice.call(arguments);t.unshift(e);analytics.push(t);return analytics}};for(var e=0;e<analytics.methods.length;e++){var key=analytics.methods[e];analytics[key]=analytics.factory(key)}analytics.load=function(key,e){var t=document.createElement("script");t.type="text/javascript";t.async=!0;t.src="https://cdn.segment.com/analytics.js/v1/" + key + "/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(t,n);analytics._loadOptions=e};analytics._writeKey="GHqd7Vfs060yZBWOEGV4ajz3S3QHYKhk";analytics.SNIPPET_VERSION="4.13.2";
```

Comment:
Does this change (often)? Would it be better moved to an actual JavaScript file?

Reply:
This is Segment's snippet which we use verbatim from them.

Reply:
Well potentially we could punt it to tag manager instead but for the time being this is fine, changing it is pretty easy and backporting if needed is easy too.